pkg/server/api/bundle/v1/service.go

```diff
@@ -56,7 +56,57 @@ func (s *Service) AppendBundle(ctx context.Context, req *bundle.AppendBundleRequ
 }
 
 func (s *Service) ListFederatedBundles(ctx context.Context, req *bundle.ListFederatedBundlesRequest) (*bundle.ListFederatedBundlesResponse, error) {
-	return nil, status.Errorf(codes.Unimplemented, "method ListFederatedBundles not implemented")
+	log := rpccontext.Logger(ctx)
+
+	if !(rpccontext.CallerIsLocal(ctx) || rpccontext.CallerIsAdmin(ctx) || rpccontext.CallerIsAgent(ctx)) {
+		log.Errorf("Permission denied: the caller must be local or present an admin or an active agent X509-SVID")
+		return nil, status.Errorf(codes.PermissionDenied, "the caller must be local or present an admin or an active agent X509-SVID")
+	}
+
+	listReq := &datastore.ListBundlesRequest{}
+
+	// Set pagination parameters
+	if req.PageSize > 0 {
```

Comment:
This will need to change if we introduce a default page size, but I _think_ it's ok for now?

From an API perspective, the page size represents the _maximum_ number of results the server will return, but the server *may* choose to return less, OR, decide to impose a default if the page size is zero.

The takeaway is that clients of the API need to always be prepared to receive a `next_page_token` even if they didn't set a page size. 

I don't think the API documentation is clear on this. I'll open a PR to add clarifying text.

Reply:
ok, so we must keep it until we decide for a default page size?

Reply:
Yes, I think this is ok for now.